router/connection_maker.go

```diff
@@ -15,31 +15,31 @@ const (
 )
 
 type ConnectionMaker struct {
-	ourself      *LocalPeer
-	peers        *Peers
-	port         int
-	targets      map[string]*Target
-	cmdLinePeers map[string]*net.TCPAddr
-	actionChan   chan<- ConnectionMakerAction
+	ourself     *LocalPeer
+	peers       *Peers
+	port        int
+	targets     map[string]*Target
+	directPeers map[string]*net.TCPAddr
+	actionChan  chan<- ConnectionMakerAction
 }
 
 // Information about an address where we may find a peer
 type Target struct {
-	attempting  bool          // are we currently attempting to connect there?
-	lastError   error         // reason for disconnection last time
-	tryAfter    time.Time     // next time to try this address
-	tryInterval time.Duration // backoff time on next failure
+	Attempting  bool          `json:"Attempting,omitempty"`  // are we currently attempting to connect there?
+	LastError   error         `json:"LastError,omitempty"`   // reason for disconnection last time
+	TryAfter    time.Time     `json:"TryAfter,omitempty"`    // next time to try this address
+	TryInterval time.Duration `json:"TryInterval,omitempty"` // backoff time on next failure
```

Comment:
This is inconsistent with the way all the other JSON is done; surely if we want this in one place we should have it in all.

Personally, I prefer to keep JSON completely separate from the main logic of weave.